util/db/interfaces.go

```diff
@@ -31,5 +32,14 @@ import (
 type Queryable interface {
 	Prepare(query string) (*sql.Stmt, error)
 	Query(query string, args ...interface{}) (*sql.Rows, error)
+	QueryContext(ctx context.Context, query string, args ...interface{}) (*sql.Rows, error)
```

Comment:
Minor design nitpick, in the `DB` and `TX` flavor impls for the `Query` and `QueryRow` functions, I see us just using `context.Background()` internally. On the new functions `QueryContext` and `QueryRowContext`, the ctx object being passed in is the same as what would be set underneath; `context.Background`.

Maybe not worth addressing as a part of this PR, but feels like we should have a common approach/set of rules for this across the codebase. i.e. should contexts _always_ be passed into methods all the way from outer edges of code, or should we default to background everywhere (as we do often now) unless there is a specific reason for a different one? As is, this seems to diverge without a discernible reason on first read.